tasks/Task.cpp

```diff
@@ -762,11 +763,60 @@ void Task::factoryReset(Arena::IDevice* device, Arena::ISystem& system)
             online = true;
         }
         catch (GenICam::TimeoutException& ex) {
-            if (base::Time::now() - start_time >= _camera_reset_timeout.get()) {
+            if (base::Time::now() - start_time >=
+                _camera_config.get().camera_reset_timeout) {
                 LOG_ERROR_S << "Timeout waiting for camera restart." << endl;
                 throw runtime_error("Timeout waiting for camera restart.");
             }
             usleep(500000);
         }
     }
+}
+
+void Task::collectInfo()
+{
+    auto current_time = base::Time::now();
+
+    if (current_time - m_last_message >= _camera_config.get().update_info) {
+
+        LOG_INFO_S << "Setting device temperature selector to "
+                   << device_temperature_selector_name.at(
+                          _camera_config.get().temperature_selector)
+                   << endl;
+        camera_lucid::CameraInfo info_message;
+        try {
+            Arena::SetNodeValue<gcstring>(m_device->GetNodeMap(),
+                "DeviceTemperatureSelector",
+                device_temperature_selector_name
+                    .at(_camera_config.get().temperature_selector)
+                    .c_str());
```

Comment:
The config is static ... that is you have to reconfigure the component to change it. Do the setting once in configureHook

Reply:
Also ... isn't setting a symbolic value like this something that is being done quite a few times ? Maybe factor it out in a helper method ?

Reply:
Sorry I added the commit, here: https://github.com/tidewise/drivers-orogen-camera_lucid/pull/3/commits/ae5ba2eb2d967bed7c0d6819136dbac82442c009

Reply:
I did there because I also used the helper for the gain. If you want I can split them in two, part here and the rest at the other commit.